utils/errors/mapping.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2020-2025 Arm Limited or its affiliates and Contributors. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package errors
+
+import (
+	"net/http"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+)
+
+// MapErrorToHTTPResponseCode maps a response status code to a common error.
+func MapErrorToHTTPResponseCode(statusCode int) error {
+	if statusCode < http.StatusBadRequest {
+		return nil
+	}
+	switch statusCode {
+	case http.StatusBadRequest:
+		return commonerrors.ErrInvalid
+	case http.StatusUnauthorized:
+		return commonerrors.ErrUnauthorised
+	case http.StatusPaymentRequired:
+		return commonerrors.ErrUnknown
+	case http.StatusForbidden:
+		return commonerrors.ErrForbidden
+	case http.StatusNotFound:
+		return commonerrors.ErrNotFound
+	case http.StatusMethodNotAllowed:
+		return commonerrors.ErrNotFound
+	case http.StatusNotAcceptable:
+		return commonerrors.ErrUnsupported
+	case http.StatusProxyAuthRequired:
+		return commonerrors.ErrUnauthorised
+	case http.StatusRequestTimeout:
+		return commonerrors.ErrTimeout
+	case http.StatusConflict:
+		return commonerrors.ErrConflict
+	case http.StatusGone:
+		return commonerrors.ErrNotFound
+	case http.StatusLengthRequired:
+		return commonerrors.ErrInvalid
+	case http.StatusPreconditionFailed:
+		return commonerrors.ErrCondition
+	case http.StatusRequestEntityTooLarge:
+		return commonerrors.ErrTooLarge
+	case http.StatusRequestURITooLong:
+		return commonerrors.ErrTooLarge
+	case http.StatusUnsupportedMediaType:
+		return commonerrors.ErrUnsupported
+	case http.StatusRequestedRangeNotSatisfiable:
+		return commonerrors.ErrOutOfRange
+	case http.StatusExpectationFailed:
+		return commonerrors.ErrUnsupported
+	case http.StatusTeapot:
+		return commonerrors.ErrUnknown
+	case http.StatusMisdirectedRequest:
+		return commonerrors.ErrUnsupported
+	case http.StatusUnprocessableEntity:
+		return commonerrors.ErrMarshalling
+	case http.StatusLocked:
+		return commonerrors.ErrLocked
+	case http.StatusFailedDependency:
+		return commonerrors.ErrFailed
+	case http.StatusTooEarly:
+		return commonerrors.ErrUnexpected
+	case http.StatusUpgradeRequired:
+		return commonerrors.ErrUnsupported
+	case http.StatusPreconditionRequired:
+		return commonerrors.ErrCondition
+	case http.StatusTooManyRequests:
+		return commonerrors.ErrUnavailable
```

Comment:
I don't think this should mark it as unavailable. 429 is commonly used to alleviate backpressure before the service actually runs out of resources so it is not that service is unavailable, it is a request that the service wait a minute before trying again.

I think it should be easier to distinguish between this and a service actually being unavailable as the behaviour. I am not sure which one it should be though as non of them really indicate the soft deny that 429 provides.